### PR TITLE
feat(dashy): Add dermatology upgrade script to ddl-shim

### PR DESCRIPTION
**Pull request by GordonSmith**
Add DashboardGrid to marshaller

Signed-off-by: Jaman Brundage <jbrundage372@gmail.com>

## Checklist:
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit message includes a "fixes" reference if appropriate.
  - [x] The commit is signed.
- [ ] The change has been fully tested:
  - [ ] I have viewed all related gallery items
  - [ ] I have viewed all related dermatology items
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised new issues to address them separately

### CODE REVIEW

**Comment by GordonSmith** [comment]
@jbrundage please review and re-test some of the upgrade samples - thx.

**Comment by jbrundage** [comment]
@GordonSmith the widget titles are no longer displaying for the Dashy-dsp-dev debug test? I'll investigate

**Comment by GordonSmith** [comment]
Screenshot?

File: packages/ddl-shim/src/dermatology.ts [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+            dv.visualization.properties.__class = "marshaller_VizChartPanel";
+            if (!dv.visualization.properties.widget) dv.visualization.properties.widget = {};
+            dv.visualization.properties.widget.__class = dv.visualization.__class;
+
+            function swap_with_supported_class(_class: string): string {
+                return classMappings[_class] ? classMappings[_class] : _class;
+            }
+        }
+        function apply_panel_property_mapping(dermPanelProps: any, dermWidgetProps: any) {
+            dv.visualization.title = dermPanelProps.title || "";
```
Comment (jbrundage) [inline packages/ddl-shim/src/dermatology.ts]: This line should be replaced with:
`if(dv.visualization.title === "")dv.visualization.title = dermPanelProps.title || "";`

File: packages/ddl-shim/src/dermatology.ts [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+                    }
+
+                }
+            }
+            if (dermWidgetProps && dermWidgetProps.showLegend && dv.visualization.properties) {
+                dv.visualization.properties.legendVisible = true;
+            }
+        }
+        function apply_widget_property_mapping(dermPanelProps: any, dermWidgetProps: any) {
+            dv.visualization.title = dermWidgetProps.title ? dermWidgetProps.title : "";
```
Comment (jbrundage) [inline packages/ddl-shim/src/dermatology.ts]: This line should be replaced with:
`if(dv.visualization.title === "")dv.visualization.title = dermWidgetProps.title || "";`

**Pull request merged**